src/Gui/View3DInventorViewer.cpp

```diff
@@ -2773,7 +2773,38 @@ void View3DInventorViewer::setCameraType(SoType t)
     }
 }
 
-void View3DInventorViewer::moveCameraTo(const SbRotation& orientation, const SbVec3f& position)
+
+namespace Gui {
+    class CameraAnimation : public QVariantAnimation
+    {
+        SoCamera* camera;
+        SbRotation startRot, endRot;
+        SbVec3f startPos, endPos;
+
+    public:
+        CameraAnimation(SoCamera* camera, const SbRotation& rot, const SbVec3f& pos)
+            : camera(camera), endRot(rot), endPos(pos)
+        {
+            startPos = camera->position.getValue();
+            startRot = camera->orientation.getValue();
+        }
+        ~CameraAnimation() override = default;
+    protected:
+        void updateCurrentValue(const QVariant & value) override
+        {
+            int steps = endValue().toInt();
+            int curr = value.toInt();
+
+            float s = static_cast<float>(curr)/static_cast<float>(steps);
+            SbVec3f curpos = startPos * (1.0f-s) + endPos * s;
+            SbRotation currot = SbRotation::slerp(startRot, endRot, s);
+            camera->orientation.setValue(currot);
+            camera->position.setValue(curpos);
+        }
+    };
+}
+
+void View3DInventorViewer::moveCameraTo(const SbRotation& rot, const SbVec3f& pos, int steps, int ms)
```

Comment:
@chennes, why was this added again? I think you tried to resolve a new conflict and something went wrong here. I replaced the class CameraAnimation with the new animations so CameraAnimation was removed.

Reply:
Ah, sorry, you're right -- I had it backwards. Can you fix?

Reply:
Yes, I can fix it, but not until the end of this week.